src/imitation/algorithms/bc.py

```diff
@@ -5,59 +5,21 @@
 """
 
 import collections
-import os
-from typing import Any, Callable, Dict, List, Mapping, Optional, Type, Union
+from typing import Any, Callable, Dict, Mapping, Optional, Tuple, Type, Union
 
-import cloudpickle
 import gym
 import numpy as np
-import tensorflow as tf
-from stable_baselines import logger
-from stable_baselines.common.policies import ActorCriticPolicy, BasePolicy
+import torch as th
+from stable_baselines3.common import logger
+from stable_baselines3.common.base_class import BaseAlgorithm
+from stable_baselines3.common.policies import ActorCriticPolicy
+from stable_baselines3.common.utils import get_schedule_fn
 from tqdm.autonotebook import trange
 
 from imitation.data import dataset, types
 from imitation.policies.base import FeedForward32Policy
 
 
-def set_tf_vars(
-    *,
-    values: List[np.ndarray],
-    scope: Optional[str] = None,
-    tf_vars: Optional[List[tf.Variable]] = None,
-    sess: Optional[tf.Session] = None,
-):
-    """Set a collection of variables to take the values in `values`.
-
-    Variables can be either specified by scope or passed directly into the
-    function as a list. Variables and values will be matched based on the order
-    in which they appear in their respective collections, so there must be as
-    many values as variables.
-
-    Args:
-        values: list of values to load into variables.
-        scope: scope to collect variables from. Either this argument xor
-          `tf_vars` must be given.
-        tf_vars: explicit list of TF variables to write to. Mutex with `scope`.
-        sess: TF session to use, if not the default.
-    """
-    if scope is not None:
-        assert tf_vars is None, "must give either `tf_vars` xor `scope` kwargs"
-        tf_vars = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES, scope=scope)
-    else:
-        assert tf_vars is not None, "must give either `tf_vars` xor `scope` kwargs"
-    assert len(tf_vars) == len(values), (
-        f"{len(tf_vars)} tf variables but " f"{len(values)} values supplied"
-    )
-    sess = sess or tf.get_default_session()
-    assert sess is not None, "must supply session or have one in context"
-    placeholders = [tf.placeholder(shape=v.shape, dtype=v.dtype) for v in tf_vars]
-    assign_ops = [tf.assign(var, ph) for var, ph in zip(tf_vars, placeholders)]
-    sess.run(
-        assign_ops, feed_dict={ph: value for ph, value in zip(placeholders, values)}
-    )
-
-
 class BC:
     def __init__(
         self,
```

Comment:
Now that we're using Torch, it might make sense to pass `policy` into `BC` directly, rather than passing in `policy_class` and `policy_kwargs` and having `BC` instantiate the policy for the user.

Reply:
Once you do that, it would also make sense to remove the `load()` method (it shouldn't be necessary when the user can directly supply a policy with the desired weights).

Reply:
Agreed! Per our discussion, this is now a TODO in the code